database.go

```diff
@@ -158,41 +163,74 @@ type datastore struct {
 var _ writestore = &datastore{}
 
 func (db *datastore) now() string {
-	if db.driverName == driverSQLite {
+	switch db.driverName {
+	case driverSQLite:
 		return "strftime('%Y-%m-%d %H:%M:%S','now')"
+	case driverMySQL, driverPostgres:
+		return "NOW()"
 	}
-	return "NOW()"
+
+	return "" // placeholder
 }
 
 func (db *datastore) clip(field string, l int) string {
-	if db.driverName == driverSQLite {
+	switch db.driverName {
+	case driverSQLite:
 		return fmt.Sprintf("SUBSTR(%s, 0, %d)", field, l)
+	case driverMySQL:
+		return fmt.Sprintf("LEFT(%s, %d)", field, l)
+	case driverPostgres:
+		return fmt.Sprintf("LEFT(%s, %d)", field, l)
 	}
-	return fmt.Sprintf("LEFT(%s, %d)", field, l)
+
+	return "" // placeholder
 }
 
 func (db *datastore) upsert(indexedCols ...string) string {
-	if db.driverName == driverSQLite {
+	cc := strings.Join(indexedCols, ", ")
+
+	switch db.driverName {
+	case driverSQLite:
 		// NOTE: SQLite UPSERT syntax only works in v3.24.0 (2018-06-04) or later
 		// Leaving this for whenever we can upgrade and include it in our binary
-		cc := strings.Join(indexedCols, ", ")
+		return "ON CONFLICT(" + cc + ") DO UPDATE SET"
+	case driverMySQL:
+		return "ON DUPLICATE KEY UPDATE"
+	case driverPostgres:
 		return "ON CONFLICT(" + cc + ") DO UPDATE SET"
 	}
-	return "ON DUPLICATE KEY UPDATE"
+
+	return "" // placeholder
 }
 
 func (db *datastore) dateAdd(l int, unit string) string {
-	if db.driverName == driverSQLite {
+	switch db.driverName {
+	case driverSQLite:
+		// NOTE: SQLite UPSERT syntax only works in v3.24.0 (2018-06-04) or later
+		// Leaving this for whenever we can upgrade and include it in our binary
 		return fmt.Sprintf("DATETIME('now', '%d %s')", l, unit)
+	case driverMySQL:
+		return fmt.Sprintf("DATE_ADD(NOW(), INTERVAL %d %s)", l, unit)
+	case driverPostgres:
+		return fmt.Sprintf("DATE_ADD(NOW(), %d %s)", l, unit)
```

Comment:
Not terribly important, but a more idiomatic way would be

```suggestion
		return fmt.Sprintf("NOW() + INTERVAL '%d %s')", l, unit)
```